lib/Runtime/Base/FunctionExecutionStateMachine.h

```diff
@@ -15,50 +15,73 @@ namespace Js
 
         // Public Getters and Setters
         ExecutionMode GetExecutionMode() const;
-        void SetExecutionMode(ExecutionMode mode); // This should eventually become private
-        ExecutionMode GetDefaultInterpreterExecutionMode() const;
+        void SetDefaultInterpreterExecutionMode();
+        void SetAsmJsExecutionMode();
+
+        // Interpreter-related functions
         ExecutionMode GetInterpreterExecutionMode(const bool isPostBailout);
         bool IsInterpreterExecutionMode() const;
         uint16 GetProfiledIterations() const;
-
         uint32 GetInterpretedCount() const { return interpretedCount; }
         uint32 SetInterpretedCount(uint32 val) { return interpretedCount = val; }
         uint32 IncreaseInterpretedCount() { return interpretedCount++; }
         bool InterpretedSinceCallCountCollection() const;
         void CollectInterpretedCounts();
         void CommitExecutedIterations();
 
+        // JIT-relatedfunctions
         void SetIsSpeculativeJitCandidate();
         uint16 GetSimpleJitLimit() const { return simpleJitLimit; }
         void ResetSimpleJitLimit();
         uint16 GetSimpleJitExecutedIterations() const;
+        void SetSimpleJitCallCount(const uint16 simpleJitLimit) const;
         void SetFullJitThreshold(const uint16 newFullJitThreshold, const bool skipSimpleJit = false);
         uint16 GetFullJitThreshold() const { return fullJitThreshold; }
         void SetFullJitRequeueThreshold(const uint16 newFullJitRequeueThreshold);
-        void SetSimpleJitCallCount(const uint16 simpleJitLimit) const;
 
         // Transition functions
         bool TryTransitionToNextExecutionMode();
         void TryTransitionToNextInterpreterExecutionMode();
         bool TryTransitionToJitExecutionMode();
         void TransitionToSimpleJitExecutionMode();
         void TransitionToFullJitExecutionMode();
-
+        
+        // Debug functions
         void PrintLimits() const;
         void AssertIsInitialized() const;
 
-
     private:
+        // This enum creates a linear view of the progression of ExecutionModes, as
+        // described in the banner comment for TryTransitionToNextExecutionMode
+        enum ExecutionState
```

Comment:
Does it make a difference for it to be int8 like executionMode was?  Consider using an enum class which avoids enum collisions.

Reply:
updated in latest commit